packages/augur-sdk/src/state/getter/Universe.ts

```diff
@@ -36,6 +58,65 @@ export class Universe {
       fees: fees.toString(),
     };
   }
+
+  @Getter('getForkMigrationTotalsParams')
+  static async getForkMigrationTotals(
+    augur: Augur,
+    db: DB,
+    params: t.TypeOf<typeof Universe.getForkMigrationTotalsParams>
+  ): Promise<MigrationTotals | NonForkingMigrationTotals> {
+    const universeForkedLogs = await db.findUniverseForkedLogs({
+      selector: {
+        universe: params.universe,
+      },
+    });
+
+    if (universeForkedLogs.length === 0) {
+      return {};
+    }
+
+    const universe = universeForkedLogs[0];
+
+    const forkingMarket = (await db.findMarkets({
+      selector: {
+        market: universe.forkingMarket,
+      },
+    }))[0];
+    const marketTypeName = marketTypeToName(forkingMarket.marketType);
+    const numTicks = Number(forkingMarket.numTicks).toString(10);
+    const minPrice = Number(forkingMarket.prices[0]).toString(10);
+    const maxPrice = Number(forkingMarket.prices[forkingMarket.prices.length - 1]).toString(10);
+
+    const children = await db.findUniverseCreatedLogs({
+      selector: {
+        parentUniverse: params.universe,
+      },
+    });
+
+    const outcomes: Outcome[] = await Promise.all(children.map(async (child): Promise<Outcome> => {
+      const payoutNumerators = child.payoutNumerators.map((hex) => Number(hex).toString(10));
+
+      // TODO min and max prices probably need to be translated into display prices
+      const outcome = calculatePayoutNumeratorsValue(maxPrice, minPrice, numTicks, marketTypeName, payoutNumerators);
+      const isMalformed = outcome === MALFORMED_OUTCOME;
+
+      const repTokenAddress = await (await augur.contracts.universeFromAddress(params.universe)).getReputationToken_();
+      const repToken = augur.contracts.reputationTokenFromAddress(repTokenAddress, augur.networkId);
+
+      return {
+        outcomeName: isMalformed ? 'malformed' : getOutcomeDescriptionFromOutcome(Number(outcome), forkingMarket),
+        outcome,
+        amount: String(await repToken.totalSupply_()),
```

Comment:
is this an off chain call? looks like an off-chain call,  if so that is frowned upon.

Reply:
It is. I talked to Justin about it - there's no sensible way to avoid this call.

It **definitely** shouldn't be inside a loop though. We only need to query it once!